### PR TITLE
Don't present Notifications on top of UIAlertController

### PR DESCRIPTION
**Pull request by yas375**
This pull request fixes an issue explained in #268

Additionally it fixes an issue when Notifications and Surveys were prevented from being presented if a notifications has been failed to be presented before.

### CODE REVIEW

**Comment by yas375** [comment]
Any feedback please?
I'd like to note that this branch also fixes a bug (see the 3rd commit) and it's proven by unit tests.
Thanks in advance!

File: Mixpanel/Mixpanel.m [hunk 1]
```diff
@@ -1596,6 +1609,10 @@ - (void)showNotificationWithObject:(MPNotification *)notification
             if (shown && ![notification.title isEqualToString:@"$ignore"]) {
                 [self markNotificationShown:notification];
             }
+
+            if (shown == NO) {
```
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: Should be `if (!shown)`.

See https://www.bignerdranch.com/blog/bools-sharp-corners/

Comment (yas375) [inline Mixpanel/Mixpanel.m]: Updated :)
(I'll read the article carefully later. thanks)

**Comment by yas375** [comment]
What do you guys think about it? I was hoping that unit tests will help to understand and verify the issue it fixes. Anything else I can do?

Thanks in advance!

**Comment by yas375** [comment]
I'd like to note that the last commit here actually fixes a bug. Do you want it to be extracted into a separate pull request?

**Comment by samgreen** [comment]
Thanks @yas375

**Comment by yas375** [comment]
:+1: thank you!

**Pull request merged**